packages/datetime/src/dateRangeInput.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as React from "react";
+
+import {
+    AbstractComponent,
+    Button,
+    Classes,
+    IInputGroupProps,
+    InputGroup,
+    Intent,
+    IProps,
+} from "@blueprintjs/core";
+
+import * as DateClasses from "./common/classes";
+import {
+    IDatePickerBaseProps,
+} from "./datePickerCore";
+
+export interface IDateRangeInputProps extends IDatePickerBaseProps, IProps {
+    startInputProps?: IInputGroupProps;
+    endInputProps?: IInputGroupProps;
+}
+
+export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, {}> {
+    public static defaultProps: IDateRangeInputProps = {};
+
+    public displayName = "Blueprint.DateRangeInput";
+
+    private startDateInputRef: HTMLInputElement = null;
+    private endDateInputRef: HTMLInputElement = null;
+
+    public render() {
+        return (
+            <div className={Classes.CONTROL_GROUP}>
+                <InputGroup
+                    className={DateClasses.DATERANGEINPUT_FIELD}
+                    inputRef={this.setStartDateInputRef}
+                    placeholder="Start date"
+                    type="text"
+                    {...this.props.startInputProps}
```

Comment:
Hmm, on second glance, I guess we'd want to combine any custom `className` contents with the `DateClasses.DATERANGEINPUT_FIELD` class above. Probably (maybe?) want to strip out the `inputRef` prop too. But does it make sense to allow developers to freely override `placeholder`, particularly with a function instead of a string? That seems like it could be nice.